lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb

```diff
@@ -134,20 +134,31 @@ def self.get_or_create_app(params)
         api_token = params[:api_token]
         owner_name = params[:owner_name]
         app_name = params[:app_name]
+        app_display_name = params[:app_display_name]
+        app_os = params[:app_os]
+        app_platform = params[:app_platform]
 
         platforms = {
           "Android" => ['Java', 'React-Native', 'Xamarin'],
           "iOS" => ['Objective-C-Swift', 'React-Native', 'Xamarin']
         }
 
+        
         if Helper::AppcenterHelper.get_app(api_token, owner_name, app_name)
           true
         else
-          if Helper.test? || UI.confirm("App with name #{app_name} not found, create one?")
-            os = Helper.test? ? "Android" : UI.select("Select OS", ["Android", "iOS"])
-            platform = Helper.test? ? "Java" : UI.select("Select Platform", platforms[os])
-
-            Helper::AppcenterHelper.create_app(api_token, owner_name, app_name, os, platform)
+          should_create_app = !app_display_name.to_s.empty? || !app_os.to_s.empty? || !app_platform.to_s.empty?
+          
+          if Helper.test? || should_create_app || UI.confirm("App with name #{app_name} not found, create one?")
+            app_display_name = app_name if app_display_name.to_s.empty?
```

Comment:
🙃 maybe `app_display_name = app_name unless app_display_name.to_s.empty` is more readable

Reply:
but it is opposite condition, isn't it? we need to set display name to name if parameter is empty

Reply:
I think you're right - got mixed in double negatives I guess.